File: CppPractice/Project1/Project1/Leetcode169.cpp
#include <string>

#include <vector>

#include <iostream>

#include <stdlib.h>

#include <time.h>

#include <sstream>

#include <algorithm>

#include <map>

#include <bitset>

using namespace std;

class Solution {
public:
    int majorityElement(vector<int>& nums) {
        map<int, int> dict;
        int count=0;
        int ans;

        for (int i = 0; i < nums.size(); i++)
        {
            dict[nums[i]]++;
        }
        for (auto x:dict)
        {
            if (x.second>count)
            {
                count = x.second;
                ans = x.first;
            }
        }
        return ans;

    }
};

File: Pythonleetcode/Leetcode169.py
class Solution(object):
    def majorityElement(self, nums):
        nums.sort()
        for i in set(nums):
            n=0
            for j in range(len(nums)):
                if i==nums[j]:
                    n=n+1
                if n>len(nums)/2:
                    return i




print(Solution().majorityElement([2,2,1,1,2,2]))